src/coreclr/debug/createdump/createdumpunix.cpp

```diff
@@ -51,23 +52,23 @@ CreateDump(const CreateDumpOptions& options)
     {
         goto exit;
     }
-    // Write the crash report json file if enabled
-    if (options.CrashReport)
+    // Write the crash report json file if enabled and createdump isn't statically linked into the runtime
+    if (options.CrashReport && !options.SingleFile && !options.NativeAOT)
```

Comment:
Make this a helper - SupportsCrashReports(Options)

Reply:
I think just explicitly checking the (new) app model enum here is just fine. A helper would only be used once (here) and to me it doesn't make it any clearer what is going on.